modules/exploits/multi/http/atlassian_crowd_pdkinstall_plugin_upload_rce.rb

```diff
@@ -0,0 +1,138 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::EXE
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Atlassian Crowd pdkinstall Unauthenticated Plugin Upload RCE',
+        'Description' => %q{
+          This module can be used to upload a plugin on Atlassian Cloud via
+          the pdkinstall development plugin as an unauthenticated attacker.
+          The payload is uploaded as a JAR archive containing a servlet using
+          a POST request to /crowd/admin/uploadplugin.action. The check command will
+          check that the /crowd/admin/uploadplugin.action page exists and that it
+          responds appropriately to determine if the target is vulnerable or not.
+        },
+        'Author' => [
+          'Paul', # Vulnerability discovery
+          'Corben Leo', # PoC and Vulnerability Writeup. @hacker_ on Twitter.
+          'Grant Willcox' # Metasploit module
+        ],
+        'License' => MSF_LICENSE,
+        'References' =>
+          [
+            ['CVE', '2019-11580'],
+            ['URL', 'https://jira.atlassian.com/browse/CWD-5388'],
+            ['URL', 'https://confluence.atlassian.com/crowd/crowd-security-advisory-2019-05-22-970260700.html'],
+            ['URL', 'https://www.corben.io/atlassian-crowd-rce/']
+          ],
+        'Platform' => %w[java],
```

Comment:
Top-level `Arch`, too, please.

Reply:
```
       Name: Atlassian Crowd pdkinstall Unauthenticated Plugin Upload RCE
     Module: exploit/multi/http/atlassian_crowd_pdkinstall_plugin_upload_rce
   Platform: Java
       Arch:
 Privileged: No
    License: Metasploit Framework License (BSD)
       Rank: Excellent
  Disclosed: 2019-05-22
```

Reply:
Eeek good spotting!

Reply:
It's a common gotcha. No worries. cc @adfoster-r7

Reply:
IOW, it won't "absorb" platform and arch from targets.

Reply:
Leaving this issue open for @adfoster-r7 to examine, however it was resolved with https://github.com/rapid7/metasploit-framework/pull/15501/commits/27f70af1b39b9b49452f727208975ed1ef518423

Reply:
@adfoster-r7